js/XAPI.js

```diff
@@ -34,6 +34,7 @@ class XAPI extends Backbone.Model {
     this.defaultLang = 'en-US';
     this.isComplete = false;
     this.changedCollectionNames = {};
+    this.retriesRemaining = this.getConfig('_retryConnectionAttempts');
```

Comment:
Sorry it's taken me so long to review this.

Instead of having an instance variable for this, which is counting up and down, because you may have multiple concurrent async statements being sent and failing, which would cause a race condition on the instance property, you should probably instead have an attempt local variable which copies the _retryConnectionAttempts at the beginning of each send?

Reply:
I haven't tested this, only reviewed the code. It looks fine to me. Please do test it before merging.